pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -193,16 +195,79 @@ func (c *Cluster) Start() error {
 	}
 	logger.Infof("wait timeout for healthy OSDs during upgrade or restart is %q", c.clusterInfo.OsdUpgradeTimeout)
 
+	osdsToSkipReconcile, err := c.getOSDsToSkipReconcile()
+	if err != nil {
+		logger.Warningf("failed to get osds to skip reconcile. %v", err)
+	}
+
 	// prepare for updating existing OSDs
 	updateQueue, deployments, err := c.getOSDUpdateInfo(errs)
 	if err != nil {
 		return errors.Wrapf(err, "failed to get information about currently-running OSD Deployments in namespace %q", namespace)
 	}
-	osdsToSkipReconcile, err := c.getOSDsToSkipReconcile()
-	if err != nil {
-		logger.Warningf("failed to get osds to skip reconcile. %v", err)
+
+	var replaceConfig *OSDReplaceConfig
+	if c.spec.Storage.Store.UpdateStore == OSDStoreUpdateConfirmation {
```

Comment:
How about moving this whole code block for replacing OSDs to a new method and just call it here? Then it keeps the `Start()` method more readable.

Reply:
done. Moved a lot of code to a new method. Looks a bit cleaner now.